### DISCUSSION TITLE
Wyze Door Pressed Notification in HA

### DISCUSSION THREAD

**Discussion by zimbrich** [SecKatie/ha-wyzeapi#393] [Q&A]
I have a Wyze Doorbell f/w 4.25.1.060 and am running the Wyze integration in HA. I only see power, siren and notifications entities for the doorbell. Notifications seems to only show state of Notifications on or off. Is it possible to sense when someone rings the Doorbell in HA, for use in an automation or script? Thanks!

**Comment by JoeSchubert** [comment]
Unfortunately, the doorbell is reported as a camera. So this integration treats it as another camera (effectively a switch). 

I recently got one of these myself so I may try to figure out how this works, assuming that it doesn't drive my wife crazy with the repeated ringing.

**Reply by mach7** [reply]
Any progress?

**Reply by EspeeFunsail** [reply]
Ant progress on this? Is there any way to let HA know when a doorbell has been pressed?

**Reply by MinnesotaMean** [reply]
Hey Joe, looking into this now too - I have a doorbell V2 that I was thinking about buying a shelly for the detect the press, but there's got to be a way the api can pass that info - as alexa and the wyze internal automation can use it. 
Did you have any luck?

**Reply by JoeSchubert** [reply]
Sorry, no I never found the time to dig into it. The last 2 years have been very busy for me with transitioning from the military into the civilian world, relocating, and everything that comes with all of that.

I will say that I doubt that there will be much to find though. Wyze typically doesn't have subscription/notifiers, the app polls the absolute hell out of their API, I imagine the official third party integrations do the same (unless they have some special API to use... But we'd have no way to MITM that to reverse engineer it). Unfortunately, we could never poll the API that hard without fear of Wyze blocking the integration (ie: the loss of binary sensor support).

The best you'd likely be able to do is poll the API to see if there's an event for it that comes back. But you'd be limited to probably every 15-ish seconds before you start generating too much API traffic for them. Which is fair, they have to pay for that traffic/processing. So it'll end up being more of a "how useful is a doorbell notification with up to a 15 second delay?"

**Comment by Ciscolay** [comment]
has there been any progress?
